django/researchdata/models.py

```diff
@@ -0,0 +1,332 @@
+    A type of relationship between Entity and Person
+    """
+
+
+class SlTypeRelEventAndPerson(SlAbstract):
+    """
+    Select List model used by RelEventAndPerson (inherits from SlAbstract model)
+    A type of relationship between Event and Person
+    """
+
+
+class SlTypeRelItemAndItem(SlAbstract):
+    """
+    Select List model used by RelItemAndItem (inherits from SlAbstract model)
+    A type of relationship between Item and Item
+    """
+
+
+class SlTypeRelItemAndPerson(SlAbstract):
+    """
+    Select List model used by RelItemAndPerson (inherits from SlAbstract model)
+    A type of relationship between Item and Person
+    """
+
+
+# Main tables
+
+
+class Entity(models.Model):
+    """
+    A collection of people, such as an organisation (e.g. a publisher, a University, etc)
+    or a division/department/centre within an organisation or a seminar/working group
+    """
+
+    name = models.CharField(max_length=255, unique=True)
+    type = models.ForeignKey(SlEntityType, on_delete=models.SET_NULL, blank=True, null=True)
+    date = models.DateField(blank=True, null=True)
+    location = models.CharField(max_length=255, blank=True, null=True, verbose_name='location (search)')
+    location_coordinates = PlainLocationField(based_fields=['location'],
+                                              zoom=7,
+                                              blank=True,
+                                              null=True,
+                                              verbose_name='location (coordinates)')
+
+    # Admin and meta fields
+    admin_notes = models.TextField(blank=True, null=True)
+    meta_created_datetime = models.DateTimeField(auto_now_add=True, verbose_name='Created')
+    meta_lastupdated_datetime = models.DateTimeField(auto_now=True, verbose_name='Last Updated')
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name_plural = 'entities'
+
+
+class Event(models.Model):
+    """
+    A meeting, conference, seminar, etc.
+    """
+
+    name = models.CharField(max_length=255, unique=True)
+    date_start = models.DateTimeField(blank=True, null=True)
+    date_end = models.DateTimeField(blank=True, null=True)
+    date_other = models.CharField(max_length=1000, blank=True, null=True)
+    type = models.ForeignKey(SlEventType, on_delete=models.SET_NULL, blank=True, null=True)
+    activity = models.ForeignKey(SlEventActivity, on_delete=models.SET_NULL, blank=True, null=True)
+    language = models.ForeignKey(SlLanguage, on_delete=models.SET_NULL, blank=True, null=True)
+
+    # Admin and meta fields
+    admin_notes = models.TextField(blank=True, null=True)
+    meta_created_datetime = models.DateTimeField(auto_now_add=True, verbose_name='Created')
+    meta_lastupdated_datetime = models.DateTimeField(auto_now=True, verbose_name='Last Updated')
+
+    def __str__(self):
+        return self.name
+
+
+class FindingAid(models.Model):
```

Comment:
I notice that your models follow the convention of being named as nouns, but this one is named more like a method (verb). Is this intentional?

Reply:
FindingAid? It's a noun, as in they refer to them "a finding aid" or "finding aids" (plural)